### PR TITLE
Paperless ngx

### PR DESCRIPTION
**Pull request by viktor-c**
I hereby request to add paperless_ngx to the list of apps, since paperless_ng has not been developed for a while and it's original repo has been archived. 
I based the files on hello_world example and used the paperless_ngx official example [docker-compose file](https://github.com/paperless-ngx/paperless-ngx/blob/main/docker/compose/docker-compose.postgres-tika.yml)

This [PR](https://github.com/davestephens/ansible-nas/pull/651) which is just a couple of hours younger as mine does not integrate tests. 

Since I am a hobby developer I could not get tests working, appreciate some help in that direction: python venv and vagrant are working; `test.sh` output is blank, `github-test-matrix.sh` outputs `::set-output name=matrix::[""]`

### CODE REVIEW

**Comment by viktor-c** [comment]
Fixed some linting errors
Setting paperless_ngx_enabled to true and then to false should trigger stopping all containers, but these tasks always get skipped in ansible ? Any idea why?

**Comment by viktor-c** [comment]
As of now, testing the role also works.

**Comment by viktor-c** [comment]
>  Setting paperless_ngx_enabled to true and then to false should trigger stopping all containers, but these tasks always get skipped in ansible ? Any idea why?

fixed that with above [commit](https://github.com/davestephens/ansible-nas/pull/652/commits/f1517976bca69ec1cec91f20fab922ebc630f56e)

**Comment by viktor-c** [comment]
Ran [CI](https://github.com/viktor-c/ansible-nas/actions/workflows/ci.yml) and [Integration](https://github.com/viktor-c/ansible-nas/actions/runs/4584935470/jobs/8096758269) (only passing for paperless-ngx).

Don't know why the other roles don't pass molecule tests, could be something to do with: https://github.com/davestephens/ansible-nas/pull/655#issue-1620010041 which fixes the tests.

**Comment by viktor-c** [comment]
[Integration](https://github.com/viktor-c/ansible-nas/actions/workflows/integration.yml)

Thanks to last Pullrequest (as of 16.04.2023) CI passes all tests.

File: roles/paperless_ngx/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+# container names
+paperless_ng_container_network_name: "paperless_ng_network"
+paperless_ng_container_name_redis: "paperless_ng_redis"
+paperless_ng_container_name_postgres: "paperless_ng_postgres"
+paperless_ng_container_name_gotenberg: "paperless_ng_gotenberg"
+paperless_ng_container_name_tika: "paperless_ng_tika"
+paperless_ng_container_name_uiserver: "paperless_ng_uiserver"
+
+paperless_tika_gotenberg_name: "gotenberg"
```
Comment (tanc) [inline roles/paperless_ngx/defaults/main.yml:54]: Should be the same as `paperless_ng_container_name_gotenberg` value, e.g. `paperless_ng_gotenberg`

File: roles/paperless_ngx/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+paperless_tika_name: "tika"
```
Comment (tanc) [inline roles/paperless_ngx/defaults/main.yml:55]: Should be the same as `paperless_ng_container_name_tika` value, e.g. `paperless_ng_tika`

**Review by tanc — CHANGES_REQUESTED** [review]
Works well except for a mismatch on the container names within the network for gotenberg and tika making then inaccessible within the network from paperless.

File: roles/paperless_ngx/defaults/main.yml
[the remark this answers is not in the document]
Comment (tanc) [reply roles/paperless_ngx/defaults/main.yml:1]: Or just `paperless_xxx`

**Pull request closed without merging**